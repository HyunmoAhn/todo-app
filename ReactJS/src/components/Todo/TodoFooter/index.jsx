import React from 'react';
import PropTypes from 'prop-types';
import cx from 'classnames';

const propTypes = {
  filter: PropTypes.string,
};
const defaultProps = {
  filter: 'all',
};

class TodoFooter extends React.Component {
  render() {
    const { filter } = this.props;
    const allClassName = cx({ selected: filter === 'all' });
    const activeClassName = cx({ selected: filter === 'active' });
    const completedClassName = cx({ selected: filter === 'completed' });

    return (
      <footer className="footer" style={{ display: 'block' }}>
        <span className="todo-count"><strong>1</strong> item left</span>
        <ul className="filters">
          <li><a href="#/" className={allClassName}>All</a></li>
          <li><a href="#/active" className={activeClassName}>Active</a></li>
          <li><a href="#/completed" className={completedClassName}>Completed</a></li>
        </ul>
        <button className="clear-completed" style={{ display: 'block' }}>
          Clear completed
        </button>
      </footer>
    );
  }
}

TodoFooter.propTypes = propTypes;
TodoFooter.defaultProps = defaultProps;

export default TodoFooter;
