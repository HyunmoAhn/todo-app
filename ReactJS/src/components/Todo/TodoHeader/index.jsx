import React from 'react';
// import PropTypes from 'prop-types';

const propTypes = {};
const defaultProps = {};

class TodoHeader extends React.Component {
  render() {
    return (
      <header className="header">
        <h1>todos</h1>
        <input className="new-todo" type="text" placeholder="What needs to be done?" />
        <input className="toggle-all" id="toggle-all" type="checkbox" />
        <label style={{ top: '15px' }} htmlFor="toggle-all">Mark all async complete</label>
      </header>
    );
  }
}

TodoHeader.propTypes = propTypes;
TodoHeader.defaultProps = defaultProps;

export default TodoHeader;
