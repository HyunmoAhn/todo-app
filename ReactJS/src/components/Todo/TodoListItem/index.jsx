import React from 'react';
import PropTypes from 'prop-types';
import cx from 'classnames';
import * as STATUS from 'constants/status';

const propTypes = {
  item: PropTypes.shape({
    value: PropTypes.string,
    id: PropTypes.string,
    status: PropTypes.string,
  }),
};
const defaultProps = {
  item: {},
};

class TodoListItem extends React.Component {
  render() {
    const { item } = this.props;
    const itemStatus = cx({
      complete: item.status === STATUS.COMPLETE,
      editing: item.status === STATUS.EDIT,
    });

    return (
      <li className={itemStatus}>
        <div className="view">
          <input type="checkbox" className="toggle" />
          <label>{item.value}</label>
          <button className="destroy" />
        </div>
        <input type="text" className="edit" />
      </li>
    );
  }
}

TodoListItem.propTypes = propTypes;
TodoListItem.defaultProps = defaultProps;

export default TodoListItem;
