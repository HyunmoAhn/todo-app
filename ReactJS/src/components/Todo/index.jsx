import React from 'react';

function Todo() {
  return (
    <section className="todoapp">
      <header className="header">
        <h1>todos</h1>
        <input className="new-todo" type="text" placeholder="What needs to be done?" />
      </header>
      <section className="main">
        <input className="toggle-all" id="toggle-all" type="checkbox" />
        <label htmlFor="toggle-all">Mark all async complete</label>
        <ul className="todo-list">
          <li>
            <div className="view">
              <input type="checkbox" className="toggle" />
              <label>value</label>
              <button className="destroy" />
            </div>
          </li>
          <li className="completed">
            <div className="view">
              <input type="checkbox" className="toggle" />
              <label>value2</label>
              <button className="destroy" />
            </div>
          </li>
          <li className="editing">
            <div className="view">
              <input type="checkbox" className="toggle" />
              <label>value2</label>
              <button className="destroy" />
            </div>
            <input type="text" className="edit" />
          </li>
        </ul>
      </section>
      <footer className="footer" style={{ display: 'block' }}>
        <span className="todo-count"><strong>1</strong> item left</span>
        <ul className="filters">
          <li><a href="#/" className="selected">All</a></li>
          <li><a href="#/active">Active</a></li>
          <li><a href="#/completed">Completed</a></li>
        </ul>
        <button className="clear-completed" style={{ display: 'block' }}>
          Clear completed
        </button>
      </footer>

    </section>
  );
}

export default Todo;
