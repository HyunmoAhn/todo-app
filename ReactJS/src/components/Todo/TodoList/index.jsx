import React from 'react';
import PropTypes from 'prop-types';
import TodoListItem from '../TodoListItem';

const propTypes = {
  todoList: PropTypes.arrayOf(
    PropTypes.shape({
      status: PropTypes.string,
      value: PropTypes.string,
      id: PropTypes.string,
    }),
  ),
  onDeleteListItem: PropTypes.func,
  onEditListItemValue: PropTypes.func,
  onEditToggleListItem: PropTypes.func,
  onToggleListItem: PropTypes.func,
};
const defaultProps = {
  todoList: [],
  onDeleteListItem() {},
  onEditListItemValue() {},
  onEditToggleListItem() {},
  onToggleListItem() {},
};

class TodoList extends React.Component {
  render() {
    const {
      todoList,
      onDeleteListItem,
      onEditListItemValue,
      onEditToggleListItem,
      onToggleListItem,
    } = this.props;

    return (
      <section className="main">
        <ul className="todo-list">
          {todoList.map(item => (
            <TodoListItem
              key={item.id}
              item={item}
              onDeleteListItem={onDeleteListItem}
              onEditListItemValue={onEditListItemValue}
              onEditToggleListItem={onEditToggleListItem}
              onToggleListItem={onToggleListItem}
            />))
          }
        </ul>
      </section>
    );
  }
}

TodoList.propTypes = propTypes;
TodoList.defaultProps = defaultProps;

export default TodoList;
