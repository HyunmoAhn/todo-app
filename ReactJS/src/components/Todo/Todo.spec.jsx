import React from 'react';
import { shallow, mount } from 'enzyme';

import Todo from './';

describe('<Todo />', () => {
  it('should match to snapshot when render default', () => {
    const wrapper = shallow(<Todo />);

    expect(wrapper).toMatchSnapshot();
  });

  it('should call onChangeFilterStatus when call handleHashChange', () => {
    const onChangeFilterStatus = jest.fn();

    const wrapper = shallow(
      <Todo
        onChangeFilterStatus={onChangeFilterStatus}
      />,
    );
    wrapper.instance().handleHashChange();

    expect(onChangeFilterStatus).toHaveBeenCalledTimes(1);
    expect(onChangeFilterStatus).toHaveBeenCalledWith('all');
  });

  it('should test window event about hashchange', () => {
    const map = {};
    window.addEventListener = jest.fn((event, cb) => {
      map[event] = cb;
    });
    window.removeEventListener = jest.fn((event) => {
      delete map[event];
    });

    const wrapper = mount(<Todo />);

    expect(map.hashchange).toEqual(expect.any(Function));

    wrapper.unmount();
    expect(map.hashchange).toEqual();
  });
});
