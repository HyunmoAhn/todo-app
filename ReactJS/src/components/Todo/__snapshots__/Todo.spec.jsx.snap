// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Todo /> should match to snapshot when render default 1`] = `
<section
  className="todoapp"
>
  <TodoHeader
    isAllCompleted={false}
    onAddListItem={[Function]}
    onToggleAllListItem={[Function]}
  />
  <TodoList
    onDeleteListItem={[Function]}
    onEditListItemValue={[Function]}
    onEditToggleListItem={[Function]}
    onToggleListItem={[Function]}
    todoList={Array []}
  />
  <TodoFooter
    completedIds={Array []}
    filter="all"
    numberOfActiveItem={0}
    onClearCompletedItems={[Function]}
  />
</section>
`;
